lib/ecto/adapters/sql/sandbox.ex

```diff
@@ -373,6 +373,43 @@ defmodule Ecto.Adapters.SQL.Sandbox do
     end
   end
 
+  @doc """
+  Starts a process that owns the connection and returns it's pid.
+
+  In tests, this is useful to ensure the pool is terminated according to the
+  order of other possible `on_exit` callbacks.
+
+      setup tags do
+        pid = Ecto.Adapters.SQL.Sandbox.start_owner!(MyApp.Repo, shared: not tags[:async])
+        on_exit(fn -> Ecto.Adapters.SQL.Sandbox.stop_owner(pid) end)
+        :ok
+      end
+
+  ## Options
+
+    * `:shared` - if `true`, the pool runs in the shared mode. Defaults to `false`
+
+  The remaining options are passed to `checkout/2`.
+  """
+  @doc since: "3.4.4"
+  def start_owner!(repo, opts \\ []) do
+    parent = self()
+
+    {:ok, pid} =
+      Agent.start(fn ->
+        {shared, opts} = Keyword.pop(opts, :shared, false)
+        :ok = checkout(repo, opts)
+
+        if shared do
+          :ok = mode(repo, {:shared, self()})
+        else
+          :ok = allow(repo, self(), parent)
+        end
+      end)
+
+    pid
+  end
+
```

Comment:
```suggestion

  @doc """
  Stops an owner process started by `start_owner!/2`.
  """
  @doc since: "3.4.4"
  @spec stop_owner(pid()) :: :ok
  def stop_owner(pid) do
    GenServer.stop(pid)
  end
```